.github/workflows/ci.yml

```diff
@@ -51,3 +51,16 @@ jobs:
 
       - name: ${{ matrix.check }}
         run: ${{ matrix.check }}
+
+  fmt:
```

Comment:
I belive it would make sense to put this as an extra step in the previous job instead of a separate one. It depends on the rust toolchain as well as on the environment.

E.g. put the install taplo as an additional step after the rust toolchain installation and add the taplo fmt --check to the matrix.

Reply:
This would install taplo for each check.

Reply:
No, that shouldn't be the case if you put the taplo install as a separate step, just like the rust-toolchain:

```
      - name: Install Taplo
        run: cargo intall ...
```

Reply:
Do you mean like here https://github.com/scs/substrate-api-client/actions/runs/3136637660/workflow#L53 ?
Because this workflow results in installing taplo each check : https://github.com/scs/substrate-api-client/actions/runs/3136637660/jobs/5093805494

Reply:
Ah, interesting. I wasn't aware of that. No, that's not what we want. Still, it should be done after the rustup show, so I think doing it like @Niederb did here: https://github.com/integritee-network/pallets/pull/98/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f makes sense.

Reply:
They way I did it it gets executed multiple times, but the Github action I use has some caching so the installation is only 2 seconds if it is in the cache. I thought of doing it similar to @echevrier but I believed I would need to install rust on another machine... but looking at the implementation here I'm not sure anymore. Maybe this is the better approach. I still think it makes sense to use the Github action in order to have the caching.

Reply:
I see. Then I start to believe the matrix itself is the wrong approach for this.  If all steps are repeated, even though they don't need to be.. such as checkout, rust installation and now the licence check. 

I'd recommend removing the matrix then altogether and add them as simple steps instead. @clangenb do you agree?

Reply:
I also think that the matrix approach uses quite a lot of resources for a speedup that is probably very small.

Reply:
I think it's fine to run it in sequence (as is proposed now?). Installing taplo for each run IMO does not matter much, since the other jobs that run in parallel, take much longer anyway.

Reply:
As alternative, you could use the docker image of taplo: `tamasfe/taplo`. GitHub Actions allows you to run a job directly based on a docker image - that way you wouldn't have to install taplo, just pull the image.